garage/tf/models/base.py

```diff
@@ -0,0 +1,49 @@
+"""This file contains the abstraction class for models."""
+import tensorflow as tf
+
+from garage.misc.overrides import overrides
+from garage.tf.core import Parameterized
+
+
+class Model(Parameterized):
+    """Abstraction class for models."""
+
+    def __init__(self):
+        """Initialize a model."""
+        Parameterized.__init__(self)
+
+    def build_model(self):
+        """
```

Comment:
What if we want to create a target model with the same graph except for variable scope?

Reply:
How will I compose models if `build_model()` doesn't take an input?

Reply:
Good point. I forgot to add this. Thanks!